pkg/apis/hive/v1alpha1/syncset_types.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// Upsert SyncSetResourceApplyMode indicates that objects will be updated
+	// or inserted (created).
+	Upsert SyncSetResourceApplyMode = "Upsert"
```

Comment:
rename the variable to indicate what it is
s/Upsert/UpsertResourceApplyMode/
otherwise, in code that uses this API, it won't be clear what hivev1.Upsert is for.
(Same as the other values)

I would put the set of values for each type in their own const (...) block for improved readability

Use only the name of the variable and not the type in the comment (s| // Upsert SyncSetResourceApplyMode | // UpsertResourceApplyMode| )